### DISCUSSION TITLE
v1.122.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#14494] [Announcements]
# v1.122.0

## Highlights

Welcome to release `v1.122.0` of Immich. After a long release break, Immich is back and brings many new features with loads of bug fixes to enhance the user experience further and make Immich more delightful to use. Some of the highlights below:

* HDR video support in the mobile app
* Multiple URLs for machine learning service
* Automatic switching between server URLs in the mobile app
* Ability to hide users when searching in an Immich instance
* Access the most recent albums through the web navigation bar
* Custom email templates
* Automatically clean up files left behind by interrupted uploads
* More responsive hosted maps for users in Oceania
* Notable fix: swiping between videos failing on older Android devices

### HDR video support

Thanks to the tireless work of @mertalev and @shenlong-tanwen, HDR videos are displayed correctly in their full dynamic range using the native video player on each respective platform directly in Immich. We hope this feature brings you a more lively experience when viewing past videos.

> [!NOTE] 
> Some videos may appear warped when viewing. If this occurs, please sign out and sign back in. This only needs to be done once and does not apply to new app installations on 1.122.0 or later.

### Multiple URLs for machine learning service

Remote machine learning enables folks to run machine learning on a more powerful computer, like their gaming PC. However, the expectation of constant uptime of the remote PC was a pain point for users. Immich now allows specifying multiple URLs so it can fall back to other PCs, including the local machine learning service on the same server.

You can add additional URL endpoints in the [Machine Learning Settings](https://my.immich.app/admin/system-settings?isOpen=machine-learning).

![image](https://github.com/user-attachments/assets/dab4dbc9-9f4a-43d6-877a-f0cf24548fa1)

### Auto-switching server URLs in the mobile app

We are happy to bring you one of the long-requested features: the ability for the app to use the local IP address of your server while you are on your home Wi-Fi and automatically switch to alternative URLs while you are out and about, ranging from VPN URLs to public DNS addresses. 

This option is located in the newly redesigned settings page in the mobile app under the Networking section. We hope this setting will simplify the networking setup you must do at home, which was previously only achievable with a split DNS setup and a local DNS server.

> [!NOTE]
>
> This feature requires *always* granting precise location permission for the Immich app so it can read the Wi-Fi name in both foreground and background.

| Android Permission | iOS Permission
| - | - |
| ![image](https://github.com/user-attachments/assets/36583576-bdb3-49d3-91b4-cb4dac3c26d0) | ![image](https://github.com/user-attachments/assets/d73466dc-2ed6-429d-b459-3a57f26e64ed) |

| New Settings Page | Local Connection Info | External Connection Info |
| - | - | - |
|  ![image](https://github.com/user-attachments/assets/8fb97f1c-cb4c-4824-8f91-0224ca0ce979) | ![image](https://github.com/user-attachments/assets/2bc32612-7451-4455-be76-ad0b196fb071) | ![image](https://github.com/user-attachments/assets/22c98230-7cf1-4817-b918-d0ec2787a951) | 

### Ability to hide users when searching in your instance

For admins hosting a public instance who don’t want users to see each other, you can now disable that mechanism in the [Server Settings](https://my.immich.app/admin/system-settings?isOpen=server).

<img src="https://github.com/user-attachments/assets/0329527e-cf78-43ee-9cbd-345a1560ea07" />

### Access the most recent albums through the web navigation bar

We added a dropdown to quickly access the most recent albums directly through the navigation bar.

<img src="https://github.com/user-attachments/assets/adc019ee-327a-4763-94ab-b9e3e1e0281d" width="300" />

### Custom email templates

You can now modify the email template for user registration, album creation, and album updates. This can be configured with the [Email Templates](https://my.immich.app/admin/system-settings?isOpen=notifications+email) setting.

![image](https://github.com/user-attachments/assets/3d6dad5d-9aab-4fcc-99cb-bb09d849ddfb)

### Automatically clean up files left behind by interrupted uploads

Previously, when an upload process was canceled, the partial files on the server stayed in the `upload` folder as corrupted files, which wasted storage space. This release implements a mechanism that intercepts the interrupted request and cleanses the partial files.

### More responsive hosted maps for users in Oceania

We have made some changes to improve the responsiveness of the hosted maps for users located in Oceania. Users located in those areas should now see their map load 2-3x faster generally as well as seeing more consistent response times.

## Support Immich

<p align="center">

<img src="https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExbjY2eWc5Y2F0ZW56MmR4aWE0dDhzZXlidXRmYWZyajl1bWZidXZpcyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/87CKDqErVfMqY/giphy.gif" width="450" title="SUPPORT THE PROJECT!"> 

</p>

If you find the project helpful, you can support Immich by purchasing a product key at https://buy.immich.app/.

Cheers! 🍻

### 🚨 Breaking Changes

* refactor: rename searchMetadata to searchAssets by @jrasm91 in <https://github.com/immich-app/immich/pull/14151>

### 🚀 Features

* feat(server): Add publicUsers toggle for user search by @samholton in <https://github.com/immich-app/immich/pull/14330>
* feat(ml): support multiple urls by @mertalev in <https://github.com/immich-app/immich/pull/14347>
* feat(web): Album preview overview in menu by @TimVanOnckelen in <https://github.com/immich-app/immich/pull/13981>
* feat(mobile): native_video_player by @mertalev & @shenlong-tanwen in <https://github.com/immich-app/immich/pull/12104>
* feat(mobile): Auto switching server URLs by @alextran1502 in <https://github.com/immich-app/immich/pull/14437>

### 🌟 Enhancements

* feat(tiles): improve map tile responsiveness in Oceania by @zackpollard in <https://github.com/immich-app/geoshenanigans/pull/62>

* feat(server): clean up interrupted upload files by @alextran1502 in <https://github.com/immich-app/immich/pull/14265>
* feat: show delete album option in empty album by @alextran1502 in <https://github.com/immich-app/immich/pull/14271>
* refactor(mobile): refactor to use context helpers for consistency by @dvbthien in <https://github.com/immich-app/immich/pull/14235>
* feat(web): persist scroll position on navigation back to album by @caburum in <https://github.com/immich-app/immich/pull/11388>
* feat(web): dedicated view for user's usage stats by @alextran1502 in <https://github.com/immich-app/immich/pull/14348>
* feat(server): specify names for thumbnail files by @eligao in <https://github.com/immich-app/immich/pull/14425>
* feat: Notification Email Templates by @TimVanOnckelen in <https://github.com/immich-app/immich/pull/13940>

### 🐛 Bug fixes

* fix(server): use hw decoding for rkmpp w/o OpenCL if possible by @zhujunsan in <https://github.com/immich-app/immich/pull/13848>
* fix(web): cannot upload folder with more than 100 files by @alextran1502 in <https://github.com/immich-app/immich/pull/14284>
* fix(web): make custom css textarea optional by @michelheusschen in <https://github.com/immich-app/immich/pull/14303>
* fix(web): clear combobox value when selectedOption is undefined by @michelheusschen in <https://github.com/immich-app/immich/pull/14334>
* fix(web): stack preview remains visible in asset viewer by @michelheusschen in <https://github.com/immich-app/immich/pull/14341>
* fix(web): timeline issues on person page by @michelheusschen in <https://github.com/immich-app/immich/pull/14366>
* fix(web): resolve issues with user usage statistics and refactor by @michelheusschen in <https://github.com/immich-app/immich/pull/14374>
* fix: ConnectivityResult.wifi regression by @systemtester in <https://github.com/immich-app/immich/pull/14401>
* fix(server): always set transcoding device, prefer renderD\* by @mertalev in <https://github.com/immich-app/immich/pull/14455>
* fix(server): show people without thumbnails by @mertalev in <https://github.com/immich-app/immich/pull/14460>
* fix(mobile): album most recent sorting on mobile by @alessandrv in <https://github.com/immich-app/immich/pull/13766>

### 📚 Documentation

* docs: add developer notes about rootless docker setup by @R-Rudolf in <https://github.com/immich-app/immich/pull/13250>
* docs: clarify/strengthen wording of PG deployment requirements by @mmomjian in <https://github.com/immich-app/immich/pull/14294>

## New Contributors

* @R-Rudolf made their first contribution in <https://github.com/immich-app/immich/pull/13250>
* @zhujunsan made their first contribution in <https://github.com/immich-app/immich/pull/13848>
* @caburum made their first contribution in <https://github.com/immich-app/immich/pull/11388>
* @systemtester made their first contribution in <https://github.com/immich-app/immich/pull/14401>
* @eligao made their first contribution in <https://github.com/immich-app/immich/pull/14425>
* @alessandrv made their first contribution in <https://github.com/immich-app/immich/pull/13766>
* @TimVanOnckelen made their first contribution in <https://github.com/immich-app/immich/pull/13940>

**Full Changelog**: <https://github.com/immich-app/immich/compare/v1.121.0...v1.122.0>

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.122.0'>v1.122.0</a>.</em>

**Comment by benjamin051000** [comment]
Incredible work! I can't wait for the app update to show up on the App Store. Also, I didn't realize I could offload ML work to my desktop... fascinating. I'll have to give that a try.

**Comment by MickLesk** [comment]
Finally the album feature (albums in sidebar) that I've been wanting for a long time! 😊 Thanks @alextran1502 & @TimVanOnckelen !

**Comment by brianwalczak** [comment]
Awesome job guys! So glad to see the custom email templates and the album updates, they'll sure come in handy. Great work as always 👏

**Comment by Alejandro18650** [comment]
Will immich automatically delete all the half-uploaded elements?

**Reply by alextran1502** [reply]
This feature only applies to new uploads; we don't have information to handle older partial files. 

You can identify them by turning on the `Storage Template` feature and then running the `Storage Template Migration` job. After that, the leftover files in the `upload` folder are partially uploaded files

**Reply by wardwygaerts** [reply]
Nice! I just discovered my upload folder is twice as big as my actual library. Lots of leftover files...

**Reply by lukanowa** [reply]
After finished migration job can I delete all files and folders in Upload folder or only files?

**Reply by catalin-l** [reply]
> After finished migration job can I delete all files and folders in Upload folder or only files?

I'm wondering the same thing. My upload folder has 138G and I'd love to clean it.

**Reply by catalin-l** [reply]
FYI, I deleted then `upload` folder and everything works as expected. Just make sure to recreate the folder and `.immich` file in it before recreating the stack. More info about the `.immich` file [here](https://immich.app/docs/administration/system-integrity/)

**Comment by ShawnTheBeachy** [comment]
Super happy to finally have multiple server URLs! Not sure I love the idea of tying it to a Wi-Fi network name, though... I'm sure you've thought it through much more than I have, but off the top of my head it seems like it would much more reliable to switch modes by polling the local URL -- say, once every 10 seconds -- and switch to the remote URL if a response isn't received within a couple of seconds. Or perhaps use websockets to monitor the local connection?

**Reply by alextran1502** [reply]
Suppose you don't have it tied to the WiFi name. In that case, the app will send the authentication token to a server set up on a random coffee's WiFi, which can hypothetically be modified to accept any random access token and intercept the photos you send there. It is for enhancing security purposes

**Reply by alextran1502** [reply]
If you have multiple AP at home with different names, you can also copy the local IP to the external URLs list. If you deem the risk is low enough for your tolerance

**Reply by ShawnTheBeachy** [reply]
@alextran1502 Makes sense. I wonder how Plex is managing it? Ah well, the currently implementation is good enough for now, I'm sure. Appreciate having it, no matter how it works 😅.

**Reply by ShawnTheBeachy** [reply]
> If you have multiple AP at home with different names, you can also copy the local IP to the external URLs list. If you deem the risk is low enough for your tolerance

Oh, I missed the multiple external URLs option on the screenshot.

**Comment by Snickers333** [comment]
It's wonderful to see the HDR videos working. Great work guys!

Are you planning on implementing HDR photos support for iOS and android in the future? On android i believe this feature is called HDR Ultra, if i'm not mistaken. Not sure how apple called their version of this feature.

Thanks a lot and again - Amazing work!

**Reply by mertalev** [reply]
We will add support for HDR photos in the future. However, I think there will be some technical challenges to it compared to videos, so it will take some development effort.

For more context: HDR support for videos was done by moving the video rendering to native iOS/Android implementations. This means that the app cannot interact with the player directly and has to use "messages" that are slower. The rendering of the UI on top of the content is also more complex and can introduce more latency. This is not as much of an issue for videos since they're less interactive by nature, but when pinching or panning an image, even a slight delay could be noticeable.

**Reply by CiNcH83** [reply]
Thought that video_player uses native implementations as well. From their docs:

> On iOS and macOS, the backing player is [AVPlayer](https://developer.apple.com/documentation/avfoundation/avplayer). The supported formats vary depending on the version of iOS, [AVURLAsset](https://developer.apple.com/documentation/avfoundation/avurlasset) class has [audiovisualTypes](https://developer.apple.com/documentation/avfoundation/avurlasset/1386800-audiovisualtypes?language=objc) that you can query for supported av formats.
On Android, the backing player is [ExoPlayer](https://google.github.io/ExoPlayer/), please refer [here](https://google.github.io/ExoPlayer/supported-formats.html) for list of supported formats.

As for native_video_player:

> On iOS, the video is displayed using a combination of [AVPlayer](https://developer.apple.com/documentation/avfoundation/avplayer) and [AVPlayerLayer](https://developer.apple.com/documentation/avfoundation/avplayerlayer).
On Android, the video is displayed using a combination of [MediaPlayer](https://developer.android.com/guide/topics/media/mediaplayer) and [VideoView](https://developer.android.com/reference/android/widget/VideoView).

Both, MediaPlayer and ExoPlayer use MediaCodec to the bottom. I can't see why ExoPlayer in video_player shouldn't support HDR. Or do they use some old version?


Tried the new app on iOS. HDR seems to work fine. Seeking is not as smooth as with video_player though.

**Reply by mertalev** [reply]
The difference is in how the video is rendered. video_player copies the video frame as a texture over to Flutter's rendering engine. native_video_player uses the native renderer and stacks content on top of it with Flutter.

The library only exposes seeking in seconds. I don't see a reason why it couldn't work with milliseconds, so we can likely change that.

**Comment by wardwygaerts** [comment]
> Access the most recent albums through the web navigation bar

Is it correct that it shows the most recent _**created**_ albums? I prefer the most recent albums based on **_date/time_** .

Maybe an option in the future to let the use choose.

**Reply by Nardo86** [reply]
I agree, I still have to recreate the old albums I had on Google Photos but in the meantime the new albums are already there... now the old ones will have priority only because I created them more recently

**Comment by shenyan-008** [comment]
I'm thinking it's time to stop using Immich.  
After updating to v1.122.0, all my photos were lost, and they all show an error when loading.  
As soon as I received the v1.122.0 update, I attempted to update immediately, but this operation caused nearly 1 million of my photos to be lost.  
I then spent several hours trying to recover them, but I failed.  
This is a problem I've never encountered since I started using Immich, and the immich-server automatically terminates after deployment.  
I tried restarting it multiple times, all to no avail.  
After several hours of attempts, I found that I could only use Immich normally by rebuilding a completely new immich-server, which made me very frustrated and disappointed.  
This is an unprecedented huge problem, making the time and money I spent on this project meaningless.  
Now I'm very disappointed, and I'm sorry, but I don't want to use Immich anymore.

**Reply by mertalev** [reply]
It sounds like an issue with the bind mount to me. You can create a GitHub discussion or make a help desk ticket on Discord and we'll do our best to help you solve it.

**Reply by cgjeffries** [reply]
Are the photos just not displaying or are they actually gone from disk? I would be very surprised if they were deleted from the disk entirely

**Reply by mertalev** [reply]
Do these images have non-ASCII (e.g. Chinese) characters by chance? It might actually be due to #14503.

**Reply by piyushpaliwal** [reply]
a reminder (to myself) to have a rock solid working backup strategy for family pictures.

**Reply by shenyan-008** [reply]
> Do these images have non-ASCII (e.g. Chinese) characters by chance? It might actually be due to #14503.这些图片可能偶然包含非 ASCII 字符（例如中文）吗？这实际上可能是因为 #14503 。

The images don't have non-ASCII characters.
And all my images won't load

**Reply by shenyan-008** [reply]
> Are the photos just not displaying or are they actually gone from disk? I would be very surprised if they were deleted from the disk entirely照片只是没有显示出来，还是实际上已经从磁盘上消失了？如果它们完全从磁盘上被删除了，我会非常惊讶

Fortunately, the images on my disk did not disappear, and I had multiple backups.
What disappeared was the time and money I spent on this library, and I couldn't recover it by using the previous version of the image.

**Reply by shenyan-008** [reply]
> It sounds like an issue with the bind mount to me. You can create a GitHub discussion or make a help desk ticket on Discord and we'll do our best to help you solve it.这听起来像是绑定挂载的问题。您可以在 GitHub 上创建一个讨论或者在 Discord 上提交一个帮助台工单，我们会尽力帮助您解决这个问题。

I want to go back to the previous version of the image but it is not available. I did not find any errors in the terminal. The image was suddenly closed automatically. I can think that the new version may have damaged the database, but I did not back up the entire database because there are too many fragmented files.

**Reply by zackpollard** [reply]
> I want to go back to the previous version of the image but it is not available. I did not find any errors in the terminal. The image was suddenly closed automatically. I can think that the new version may have damaged the database, but I did not back up the entire database because there are too many fragmented files.

We added automatic database backups a couple of versions ago. If you are running a default setup, you may have these backups still. They are located by default alongside your library in UPLOAD_LOCATION/backups. Check on disk and you may find them. Regardless, it is crucial for any application with a database to ensure you have working backups. For the future, here is some documentation on how to setup backups for your database manually https://immich.app/docs/administration/backup-and-restore.

Also, we never removed previous versions of the images, they are always left available.

**Comment by Str1atum** [comment]
Does one need to reencode the videos for HDR on mobile to work?

**Reply by alextran1502** [reply]
There is no need

**Reply by Str1atum** [reply]
But doesn't it automatically play the recoded SDR version then?

**Reply by alextran1502** [reply]
This is strictly about the mobile app using the native player, which supports HDR. The previous player doesn't support HDR content.

**Reply by mertalev** [reply]
Videos that exist locally and remote HDR videos that are not transcoded will be played in HDR. However, if you play remote videos that were transcoded, then the SDR transcode will be served regardless even if the client now supports HDR. We can think about how to improve on this.

**Reply by Snickers333** [reply]
> Videos that exist locally and remote HDR videos that are not transcoded will be played in HDR. However, if you play remote videos that were transcoded, then the SDR transcode will be served regardless even if the client now supports HDR. We can think about how to improve on this.

Even if in the transcoding options we disable tonemapping entirely?

**Reply by mertalev** [reply]
The result will still be SDR if you disable tone-mapping - the colors will just be less accurate. It's more of a debugging option than anything. Maybe it'd be better for it to not convert to SDR at all when disabled.

**Comment by FreeWind6** [comment]
After the update, I noticed that the videos made on the iPhone 15 ProMax, on Android, there is no sound on these videos. Tell me, is this a known problem?

**Reply by FreeWind6** [reply]
https://github.com/immich-app/immich/issues/14653
